docs/news.rst

```diff
@@ -29,7 +29,41 @@ Highlights:
      [transfer]
      max_threads=16
 
-  This release also includes other optimizations associated with file uploads.
+- This release includes support for directly accessing
+  `STS storage locations <https://docs.synapse.org/articles/sts_storage_locations.html>`__.
+  This allows reading and writing data to Synapse storage using external tools, such as the boto3 Python library and
+  the awscli. Currently this feature is limited to Amazon S3 storage at locations enabled for STS, and is limited to
+  read_only for Synapse storage while read_write is supported for
+  `custom S3 storage locations <https://docs.synapse.org/articles/custom_storage_location.html>`__ enabled for STS.
+
+  .. code-block::
+
+    # get an STS token for a previously enabled storage location and use with boto
+    import boto3
+    boto_credentials = synapseclient.get_sts_storage_token(folder_id, 'read_only', output_format='boto')
+    s3_client = boto3.client('s3', **boto_credentials)
+    s3_client.download_file(<bucket>, <key>, <filename>)
+
+  .. code-block::
+
+    # from the command line output commands that can be used to supply credentials to the awscli
```

Comment:
Awesome, so with this enabled, the client would check if a folder has STS enabled?   If the token is expired, would the user be prompted to get a new token?

Reply:
If the user is using boto3 transparently via the synapse client (i.e. they have an STS storage location configured and have use_boto_sts=true in their .synapseConfig) then they don’t need to manage tokens, uploads and downloads will automatically favor using boto to conduct the transfer, and tokens will be cached (in memory for the life of the process) and refreshed as needed.

If they are exporting the token to use themselves e.g. from the command line and the awscli then they will need to get a new token every 12 hours.